### PR TITLE
Create Jenkins job for terraform deployments

### PR DESCRIPTION
**Pull request by surminus**
We require a Jenkins job so we can keep track of deployments for Terraform. This commit adds the job manifest and template, and also adds in the hieradata required, including the list of projects available to
deploy.

Story: https://trello.com/c/FOG8jmb5/53-create-jenkins-job-to-deploy-aws-configuration

### CODE REVIEW

**Comment by mattbostock** [comment]
Merge when ready :+1:

**Comment by surminus** [comment]
I actually had a thought about this on the way home yesterday, instead of doing an `export` in the builder shell, I think we can actually do an inject of parameters somewhere? That would be a lot cleaner.

**Comment by mattbostock** [comment]
@surminus Which `export` are you thinking of?

File: modules/govuk_jenkins/templates/jobs/deploy_terraform_project.yaml.erb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+              - master
+- job:
+    name: Deploy_Terraform_Project
+    display-name: Deploy_Terraform_Project
+    project-type: freestyle
+    description: |
+      Deploy a specific Terraform project in <%= @environment -%>.
+    properties:
+        - github:
+          url: https://github.com/alphagov/govuk-terraform-provisioning/
```
Comment (alexmuller) [inline modules/govuk_jenkins/templates/jobs/deploy_terraform_project.yaml.erb]: I think this needs a bit of indentation of the `url` to be valid YAML.

**Comment by surminus** [comment]
@mattbostock I updated the export bit. It originally did `export DEPLOY_ENV=foo` in the builder shell, but I updated it [here](https://github.com/alphagov/govuk-puppet/blob/terraform-jenkins-job/modules/govuk_jenkins/templates/jobs/deploy_terraform_project.yaml.erb#L18-L21)

**Comment by surminus** [comment]
@alexmuller thanks, amended the yaml so hopefully it's correct now

**Comment by mattbostock** [comment]
:+1:

**Pull request merged**